File: backend/pricingService/src/main/java/com/nsbm/bunmart/pricing/services/CampaignService.java
package com.nsbm.bunmart.pricing.services;

import com.nsbm.bunmart.pricing.model.Campaign;
import com.nsbm.bunmart.pricing.interfaces.ICampaignService;
import com.nsbm.bunmart.pricing.repositories.CampaignRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
@RequiredArgsConstructor
public class CampaignService implements ICampaignService {

    private final CampaignRepository repository;

    @Override
    public Campaign create(Campaign campaign) {
        return repository.save(campaign);
    }

    @Override
    public Campaign getById(Long id) {
        return repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Campaign not found with id: " + id));
    }

    @Override
    public List<Campaign> getAll() {
        return repository.findAll();
    }

    @Override
    public Campaign update(Long id, Campaign campaign) {
        Campaign existing = getById(id);
        existing.setName(campaign.getName());
        existing.setDescription(campaign.getDescription());
        existing.setStartDate(campaign.getStartDate());
        existing.setEndDate(campaign.getEndDate());
        existing.setIsActive(campaign.getIsActive());
        return repository.save(existing);
    }

    @Override
    public void delete(Long id) {
        repository.deleteById(id);
    }
}

File: frontend/pricing/services/campaignService.ts
import { api } from '../../service/api';
import { Campaign, CreateCampaignDto } from '../models/Campaign';

export const campaignService = {
  getAll: async (): Promise<Campaign[]> => {
    const response = await api.get('/campaigns');
    return response.data;
  },

  getById: async (id: number): Promise<Campaign> => {
    const response = await api.get(`/campaigns/${id}`);
    return response.data;
  },

  create: async (data: CreateCampaignDto): Promise<Campaign> => {
    const response = await api.post('/campaigns', data);
    return response.data;
  },

  update: async (id: number, data: Partial<CreateCampaignDto>): Promise<Campaign> => {
    const response = await api.put(`/campaigns/${id}`, data);
    return response.data;
  },

  delete: async (id: number): Promise<void> => {
    await api.delete(`/campaigns/${id}`);
  },
};
